mav_trajectory_generation/src/trajectory.cpp

```diff
@@ -255,6 +259,14 @@ bool Trajectory::addTrajectories(const std::vector<Trajectory>& trajectories,
   return true;
 }
 
+bool Trajectory::offsetTrajectory(const Eigen::VectorXd& A_r_B) {
+  for (Segment& s : segments_) {
+    if (!s.offsetSegment(A_r_B)) return false;
```

Comment:
Warning, nitpick:
Could lead to weird results if not all segments are transformed. 
I know, based on the code in offsetSegment this should not happen, as A_r_B either works for all or none of the segments. But it's more convoluted to read, as it's not necessary.

How about adding the same check for dimension here and just do the check once in the beginning of the function? Then it's more clear what happens.